amlb/datasets/file.py

```diff
@@ -129,6 +138,38 @@ def __repr__(self):
         return repr_def(self)
 
 
+    def extend_dataset_with_timeseries_config(self, dataset, dataset_config):
```

Comment:
we are manipulating the outer context of dataset_config, maybe this is ok but want to mention that it is happening.

Reply:
Same with manipulating outer context of dataset. Consider adding documentation stating that this is intended or otherwise do a deep copy on the objects.